src/config.rs

```diff
@@ -82,29 +128,28 @@ impl Config {
 
     fn filter_based_on_rule_argument(
         &self,
-        active_rules: Vec<Arc<dyn Rule>>,
-    ) -> Vec<Arc<dyn Rule>> {
+        active_rules_and_associated_plugins: Vec<(Arc<dyn Rule>, Option<PluginIndex>)>,
+    ) -> Vec<(Arc<dyn Rule>, Option<PluginIndex>)> {
```

Comment:
So for these couple internal methods we're passing around these slightly clunky `(Arc<dyn Rule>, Option<PluginIndex>)` tuples (while we're trying to "get organized" as far as which rules are active etc and associated with which plugin)